homeassistant/components/viaris/sensor.py

```diff
@@ -0,0 +1,1207 @@
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state=get_total_power,
+    ),
+    ViarisSensorEntityDescription(
+        key=FV_POWER_KEY,
+        icon="mdi:solar-power-variant",
+        name="Solar and battery power",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        entity_registry_enabled_default=True,
+        state=get_state_solar,
+        disabled=False,
+    ),
+    ViarisSensorEntityDescription(
+        key=TMC100_KEY,
+        name="TMC100",
+        icon="mdi:meter-electric",
+        entity_category=EntityCategory.CONFIG,
+        state=get_tmc100_detected,
+    ),
+    ViarisSensorEntityDescription(
+        key=CONTAX_D0613_KEY,
+        name="Contax D0613",
+        icon="mdi:meter-electric-outline",
+        entity_category=EntityCategory.CONFIG,
+        state=get_ctx_detected,
+    ),
+    ViarisSensorEntityDescription(
+        key=OVERLOAD_REL_KEY,
+        name="Overload rel",
+        icon="mdi:checkbox-blank-circle",
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state=get_rel_overload,
+    ),
+    ViarisSensorEntityDescription(
+        key=ACTIVE_POWER_CONN1_KEY,
+        name="Active power connector 1",
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state=get_active_power_conn1,
+    ),
+    ViarisSensorEntityDescription(
+        key=ACTIVE_POWER_CONN2_KEY,
+        name="Active power connector 2",
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state=get_active_power_conn2,
+    ),
+    ViarisSensorEntityDescription(
+        key=REACTIVE_POWER_CONN2_KEY,
+        name="Reactive power connector 2",
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state=get_reactive_power_conn2,
+    ),
+    ViarisSensorEntityDescription(
+        key=REACTIVE_POWER_CONN1_KEY,
+        name="Reactive power connector 1",
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state=get_reactive_power_conn1,
+    ),
+)
+
+
+def get_user_connector1(value) -> str:
+    """Extract user connector1."""
+    data = json_loads(value)
+    if data["data"]["name"] == "mennekes" or data["data"]["name"] == "mennekes1":
```

Comment:
The json loading and parsing should also be done by the 3rd party library. The library should provide objects with attributes that represent the sensor data.

I recommend using `pydantic` in the library to create typed objects from json data.